src/Pages/Messaging/Messaging.js

```diff
@@ -0,0 +1,169 @@
+    } else {
+      setErrorMessage('');
+    }
+
+    setSent(true);
+    setMessage('');
+
+    setTimeout(() => {
+      setSent(false);
+    }, 3000);
+  };
+
+  return (
+    <form
+      onSubmit={handleSubmit}
+      className="w-1/2 mx-auto p-6 bg-zinc-900 rounded-md shadow-lg"
+    >
+      <div className="mb-4">
+        <label htmlFor="room" className="block text-sm font-medium text-gray-400">
+          Room
+        </label>
+        <input
+          autoComplete="off"
+          id="room"
+          type="text"
+          value={id}
+          onChange={(e) => setId(e.target.value)}
+          className="mt-1 block w-full px-3 py-2 border border-gray-300 rounded-md shadow-sm focus:outline-none focus:ring-blue-500 focus:border-blue-500"
+          placeholder="Enter room name"
+        />
+      </div>
+
+      <div className="mb-4">
+        <label htmlFor="key" className="block text-sm font-medium text-gray-400">
+          Key
+        </label>
+        <input
+          autoComplete="off"
+          id="key"
+          type="text"
+          value={apiKey}
+          onChange={(e) => setApiKey(e.target.value)}
+          className="mt-1 block w-full px-3 py-2 border border-gray-300 rounded-md shadow-sm focus:outline-none focus:ring-blue-500 focus:border-blue-500"
+          placeholder="Enter your key"
+        />
+      </div>
+
+      <div className="mb-4">
+        <label htmlFor="message" className="block text-sm font-medium text-gray-400">
+          Message
+        </label>
+        <textarea
+          id="message"
+          autoComplete="off"
+          value={message}
+          onChange={(e) => setMessage(e.target.value)}
+          className="mt-1 block w-full px-3 py-2 border border-gray-300 rounded-md shadow-sm focus:outline-none focus:ring-blue-500 focus:border-blue-500"
+          placeholder="Enter your message"
+        ></textarea>
+      </div>
+
+      <button
+        type="submit"
+        className="w-full px-4 py-2 text-white bg-blue-500 rounded-md shadow-md hover:bg-blue-600 focus:outline-none focus:ring-2 focus:ring-blue-500 focus:ring-offset-2"
+      >
+        Submit
+      </button>
+
+      {sent && (
+        errorMessage ? (
+          <div className="mt-4 p-2 min-h-12 flex items-center justify-center bg-red-100 text-red-700 rounded-md shadow-md">
+            {errorMessage}
+          </div>
+        ) : (
+          <div className="mt-4 p-2 min-h-12 flex items-center justify-center bg-green-100 text-green-700 rounded-md shadow-md">
+            Message sent to room {`'${id || 'general'}'`}
+          </div>
+        )
+      )}
+    </form>
+  );
+}
+
+function Feed(props) {
+  const { token } = props;
+  const [messages, setMessages] = useState([]);
+  const [apiKey, setApiKey] = useState('');
+  const [error, setError] = useState('');
+  const roomId = 'general';
+  let eventSource;
+
+  useEffect(() => {
+    const fetchApiKey = async () => {
```

Comment:
how about for sending messages, we allow the user to pass in a token, that way we dont need to handle api key logic in here


api keys can be the second option that our messaging apis use, if a token wasnt sent, but an api key was, let me know if this makes sense

by allowing tokens, we dont need this function

Reply:
lemme handle this

Reply:
implemented, probably need to check whether the solution is clean or not